core/src/test/java/io/questdb/cutlass/line/tcp/AlterTableLineTcpReceiverTest.java

```diff
@@ -123,6 +121,116 @@ public void testAlterCommandDropPartition() throws Exception {
+
+                    try (TableModel tm = new TableModel(configuration, "plug", PartitionBy.DAY)) {
+                        tm.col("room", ColumnType.SYMBOL);
+                        tm.col("watts", ColumnType.LONG);
+                        tm.timestamp();
+
+                        CairoTestUtils.create(tm);
+                    }
+                    try (TableWriter writer = engine.getWriter(AllowAllCairoSecurityContext.INSTANCE, "plug", "test")) {
+                        TableWriter.Row row = writer.newRow(day1 / 1000);
+                        row.putSym(0, "6A");
+                        row.putLong(1, 100L);
+                        row.append();
+                        writer.commit();
+                    }
+
+                    Assert.assertNull(sendWithAlterStatement(
+                            server,
+                            "plug,room=6A watts=1i " + day1 + "\n" +
+                                    "plug,room=6B watts=37i " + day1 + "\n" +
+                                    "plug,room=7G watts=21i " + day1 + "\n" +
+                                    "plug,room=1C watts=11i " + day1 + "\n",
+                            WAIT_ALTER_TABLE_RELEASE,
+                            "ALTER TABLE plug DROP PARTITION LIST '2023-02-27'"));
+                    assertTable("room\twatts\ttimestamp\n");
+
+                    send(server, "plug,room=6A watts=125i " + day1 + "\n");
+                    assertTable("room\twatts\ttimestamp\n" +
+                            "6A\t125\t2023-02-27T00:00:00.000000Z\n");
+                },
+                true, 50L
+        );
+    }
+
+    @Test
+    public void testAlterCommandDropAllPartitions() throws Exception {
+        long day1 = IntervalUtils.parseFloorPartialTimestamp("2023-02-27") * 1000;
+        long day2 = IntervalUtils.parseFloorPartialTimestamp("2023-02-28") * 1000;
+        runInContext((server) -> {
+
+                    final AtomicLong ilpProducerWatts = new AtomicLong(0L);
+                    final AtomicBoolean keepSending = new AtomicBoolean(true);
+                    final SOCountDownLatch ilpProducerHalted = new SOCountDownLatch(1);
+
+                    final Thread ilpProducer = new Thread(() -> {
+                        String lineTpt = "plug,room=6A watts=\"%di\" %d%n";
+                        try {
+                            while (keepSending.get()) {
+                                try {
+                                    long watts = ilpProducerWatts.getAndIncrement();
+                                    long day = (watts + 1) % 4 == 0 ? day1 : day2;
+                                    String lineData = String.format(lineTpt, watts, day);
+                                    send(server, lineData);
+                                    LOG.info().$("sent: ").$(lineData).$();
+                                } catch (Exception e) {
+                                    Assert.fail("could not send");
```

Comment:
test will not fail of thread fails. It will print stack trace but test will pass

Reply:
arrgh, well spotted, thank you.